src/main/java/com/codejoust/main/dto/user/UserMapper.java

```diff
@@ -15,7 +16,18 @@ public static UserDto toDto(User entity) {
         if (entity == null) {
             return null;
         }
-        return mapper.map(entity, UserDto.class);
+        UserDto userDto = mapper.map(entity, UserDto.class);
+
+        // Attach the account to the user, if it exists.
+        if (entity.getAccount() != null) {
+            AccountUidDto accountUidDto = new AccountUidDto();
+            accountUidDto.setUid(entity.getAccount().getUid());
+            userDto.setAccountUid(accountUidDto);
+        } else {
+            userDto.setAccountUid(null);
+        }
```

Comment:
I was surprised too, but it is necessary - I'm not sure how the `mapper.map` algorithm works, but the screenshot of the debugging process below shows that the `accountUid` is not yet set.

**Debugging the toDto Method**
<img width="1138" alt="Debugging the toDto Method" src="https://user-images.githubusercontent.com/7987279/122629847-cf70ec00-d074-11eb-9670-d837f7899b16.png">

Reply:
I see - looking at it I think it's actually cause the account field is named accountUid instead of account in the UserDto so it can't map automatically. This isn't a major deal either way tho if you prefer to leave this as is

Reply:
Ahh ok - I'll try to make that change, as this would simplify things quite a bit 👍

Reply:
Done! ✅